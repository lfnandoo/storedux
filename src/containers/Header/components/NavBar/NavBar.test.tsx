import renderer from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react";

import { NavBar } from "./index";

describe("NavBar", () => {
  const onClearAll = jest.fn();
  const amount = 2;

  it("should render correctly", () => {
    const tree = renderer
      .create(<NavBar itemsAmount={amount} onClearAll={onClearAll} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should get items amount", () => {
    const { getByTestId } = render(
      <NavBar itemsAmount={amount} onClearAll={onClearAll} />
    );

    expect(getByTestId("items-amount")).toHaveTextContent(amount.toString());
  });

  it("should trigger onClearAll", () => {
    const { getByTestId } = render(
      <NavBar itemsAmount={amount} onClearAll={onClearAll} />
    );

    fireEvent.click(getByTestId("clear-all-button"));

    expect(onClearAll).toHaveBeenCalled();
  });
});
