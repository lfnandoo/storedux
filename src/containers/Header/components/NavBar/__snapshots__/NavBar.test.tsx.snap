// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render correctly 1`] = `
<nav
  className="sc-gtsrHT bDpdCY"
>
  <h1>
    STO
    <strong>
      REDUX
    </strong>
  </h1>
  <div
    className="sc-dlnjwi AwWBE"
  >
    <div>
      <h3>
        Itens: 
        <strong
          data-testid="items-amount"
        >
          2
        </strong>
      </h3>
    </div>
    <button
      className="sc-bdnxRM OPJfk"
      data-testid="clear-all-button"
      onClick={[MockFunction]}
    >
      Limpar tudo
    </button>
  </div>
</nav>
`;
